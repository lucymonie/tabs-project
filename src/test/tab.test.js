import React from 'react'
import { shallow } from 'enzyme'
import Tab from '../components/tab'

const props = {
  section: 'technology',
  onClick: jest.fn()
}

const component = shallow(<Tab {...props} />)

it('renders a tab', () => {
  expect(component).toMatchSnapshot()
})

it('renders a list item', () => {
  const li = component.find('li')
  expect(li).toHaveLength(1)
})

it('renders an anchor text with section text', () => {
  const a = component.find('a')
  expect(a).toHaveLength(1)
  expect(a.text()).toBe('technology')
})

it('registers a click event', () => {
  const li = component.find('li')
  li.simulate('click')
  expect(props.onClick.mock.calls).toHaveLength(1)
})
