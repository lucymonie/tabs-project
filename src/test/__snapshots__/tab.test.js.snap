// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a tab 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tab
    onClick={[MockFunction]}
    section="technology"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="tabs-style"
        href="#technology"
      >
        technology
      </a>,
      "className": "tab col s4",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "technology",
        "className": "tabs-style",
        "href": "#technology",
      },
      "ref": null,
      "rendered": "technology",
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="tabs-style"
          href="#technology"
        >
          technology
        </a>,
        "className": "tab col s4",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "technology",
          "className": "tabs-style",
          "href": "#technology",
        },
        "ref": null,
        "rendered": "technology",
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
